src/App.tsx

```diff
@@ -1,39 +1,56 @@
 import React from 'react';
+
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+export interface State {
+  hasClock: boolean,
+  clockName: string,
+}
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
```

Comment:
Checklist p1 and p4
![image](https://github.com/mate-academy/react_clock/assets/46566190/83312f3c-d562-4979-a766-8a7c6b0c69f6)
![image](https://github.com/mate-academy/react_clock/assets/46566190/e899755e-7a94-4cc4-83c4-b57dfdeee4a7)